tests/integration/kubernetes/gha-run.sh

```diff
@@ -268,14 +268,15 @@ function run_tests() {
 	# and enable cri plugin in containerd config.
 	# TODO: enable testing auto-generated policy for other types of hosts too.
 
-	if [ "${KATA_HOST_OS}" = "cbl-mariner" ]; then
+	if [ "${KATA_HOST_OS}" = "cbl-mariner" ] || [ "${KATA_HYPERVISOR}" = "qemu-tdx" ] ; then
 
 		export AUTO_GENERATE_POLICY="yes"
-
-		# set default containerd config
-		sudo containerd config default | sudo tee /etc/containerd/config.toml > /dev/null
-		echo "containerd config has been set to default"
-		sudo systemctl restart containerd && sudo systemctl is-active containerd
+		if [ "${GENPOLICY_PULL_METHOD}" = "containerd" ]; then
+			# set default containerd config
+			sudo containerd config default | sudo tee /etc/containerd/config.toml > /dev/null
+			echo "containerd config has been set to default"
+			sudo systemctl restart containerd && sudo systemctl is-active containerd
```

Comment:
Same here about the restart ...

For k3s, you'd need to restart `k3s.service`.

Please, take a look at https://github.com/kata-containers/kata-containers/blob/main/tools/packaging/kata-deploy/scripts/kata-deploy.sh

There you'll see that we deal with k3s, k0s, and rke2 apart from vanilla k8s, and you need to take that into consideration here.

For the paths for configs, take a look at each one of the overlays: https://github.com/kata-containers/kata-containers/tree/main/tools/packaging/kata-deploy/kata-deploy/overlays